plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveTimestampPrecisionOnHiveViews.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.Session;
+import io.trino.plugin.hive.containers.HiveHadoop;
+import io.trino.plugin.hive.containers.HiveMinioDataLake;
+import io.trino.plugin.hive.s3.S3HiveQueryRunner;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestHiveTimestampPrecisionOnHiveViews
```

Comment:
looks that there already was test coverage (plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java & product tests). Do we need this new class? What is it adding?

Reply:
`io.trino.plugin.hive.BaseHiveConnectorTest#testSelectFromPrestoViewReferencingHiveTableWithTimestamps` does not have  a test containing catalogs with different `hive.timestamp-precision` properties.

`io.trino.tests.product.hive.AbstractTestHiveViews#testTimestampHiveView` OTOH has indeed everything what is needed to cover the changes from this PR.

Thank you for pointing this out.